_analyzers/token-filters/edge-ngram.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: default
+title: Edge ngram
+parent: Token filters
+nav_order: 120
+---
+<!-- vale off -->
+# Edge ngram token filter
+<!-- vale on -->
+The `edge_ngram` token filter generates n-grams (substrings) from the beginning (edge) of a token. It's particularly useful in scenarios like autocomplete or prefix matching, where you want to match the start of words or phrases as the user types.
+
```

Comment:
Could you add a sentence about the difference between this token filter and the `ngram` token filter? We should also link to the `ngram` token filter when it is merged and also link to the search as you type page because this functionality is useful for that (https://opensearch.org/docs/2.17/search-plugins/searching-data/autocomplete/#search-as-you-type).